File: packages/expo-contacts/android/src/main/java/expo/modules/contacts/models/PhoneNumberModel.java
package expo.modules.contacts.models;

import android.database.Cursor;
import android.provider.ContactsContract;

import java.util.Map;

import expo.modules.contacts.EXColumns;

import static android.provider.ContactsContract.*;

public class PhoneNumberModel extends BaseModel {
    @Override
    public String getContentType() {
        return CommonDataKinds.Phone.CONTENT_ITEM_TYPE;
    }

    @Override
    public String getDataAlias() {
        return "number";
    }

    @Override
    public void fromMap(Map<String, Object> readableMap) {
        super.fromMap(readableMap);

        String phoneNumber = getData();
        map.putString("digits", phoneNumber.replaceAll("[^\\d.]", ""));
    }

    @Override
    public int mapStringToType(String label) {
        int phoneType;
        switch (label) {
        case "home":
            phoneType = CommonDataKinds.Phone.TYPE_HOME;
            break;
        case "mobile":
            phoneType = CommonDataKinds.Phone.TYPE_MOBILE;
            break;
        case "work":
            phoneType = CommonDataKinds.Phone.TYPE_WORK;
            break;
        case "faxWork":
            phoneType = CommonDataKinds.Phone.TYPE_FAX_WORK;
            break;
        case "faxHome":
            phoneType = CommonDataKinds.Phone.TYPE_FAX_HOME;
            break;
        case "pager":
            phoneType = CommonDataKinds.Phone.TYPE_PAGER;
            break;
        case "callback":
            phoneType = CommonDataKinds.Phone.TYPE_CALLBACK;
            break;
        case "car":
            phoneType = CommonDataKinds.Phone.TYPE_CAR;
            break;
        case "companyMain":
            phoneType = CommonDataKinds.Phone.TYPE_COMPANY_MAIN;
            break;
        case "isdn":
            phoneType = CommonDataKinds.Phone.TYPE_ISDN;
            break;
        case "main":
            phoneType = CommonDataKinds.Phone.TYPE_MAIN;
            break;
        case "otherFax":
            phoneType = CommonDataKinds.Phone.TYPE_OTHER_FAX;
            break;
        case "radio":
            phoneType = CommonDataKinds.Phone.TYPE_RADIO;
            break;
        case "telex":
            phoneType = CommonDataKinds.Phone.TYPE_TELEX;
            break;
        case "ttyTdd":
            phoneType = CommonDataKinds.Phone.TYPE_TTY_TDD;
            break;
        case "workMobile":
            phoneType = CommonDataKinds.Phone.TYPE_WORK_MOBILE;
            break;
        case "workPager":
            phoneType = CommonDataKinds.Phone.TYPE_WORK_PAGER;
            break;
        case "assistant":
            phoneType = CommonDataKinds.Phone.TYPE_ASSISTANT;
            break;
        case "mms":
            phoneType = CommonDataKinds.Phone.TYPE_MMS;
            break;
        case "other":
            phoneType = CommonDataKinds.Phone.TYPE_OTHER;
            break;
        default:
            phoneType = EXColumns.TYPE_CUSTOM;

            break;
        }
        return phoneType;
    }

    @Override
    protected String getLabelFromCursor(Cursor cursor) {
        String label = super.getLabelFromCursor(cursor);
        if (label != null)
            return label;
        switch (cursor.getInt(cursor.getColumnIndex(EXColumns.TYPE))) {
        case CommonDataKinds.Phone.TYPE_HOME:
            return "home";
        case CommonDataKinds.Phone.TYPE_WORK:
            return "work";
        case CommonDataKinds.Phone.TYPE_MOBILE:
            return "mobile";
        case CommonDataKinds.Phone.TYPE_OTHER:
            return "other";
        default:
            return "unknown";
        }
    }

}

File: packages/expo-contacts/android/src/main/java/expo/modules/contacts/models/PhoneNumberModel.kt
package expo.modules.contacts.models

import android.database.Cursor
import android.provider.ContactsContract.CommonDataKinds
import expo.modules.contacts.Columns

class PhoneNumberModel : BaseModel() {
  override val contentType: String = CommonDataKinds.Phone.CONTENT_ITEM_TYPE
  override val dataAlias: String = "number"

  override fun fromMap(readableMap: Map<String, Any?>) {
    super.fromMap(readableMap)
    val phoneNumber = data!!
    map.putString("digits", phoneNumber.replace("[^\\d.]".toRegex(), ""))
  }

  override fun mapStringToType(label: String?): Int {
    val phoneType = when (label) {
      "home" -> CommonDataKinds.Phone.TYPE_HOME
      "mobile" -> CommonDataKinds.Phone.TYPE_MOBILE
      "work" -> CommonDataKinds.Phone.TYPE_WORK
      "faxWork" -> CommonDataKinds.Phone.TYPE_FAX_WORK
      "faxHome" -> CommonDataKinds.Phone.TYPE_FAX_HOME
      "pager" -> CommonDataKinds.Phone.TYPE_PAGER
      "callback" -> CommonDataKinds.Phone.TYPE_CALLBACK
      "car" -> CommonDataKinds.Phone.TYPE_CAR
      "companyMain" -> CommonDataKinds.Phone.TYPE_COMPANY_MAIN
      "isdn" -> CommonDataKinds.Phone.TYPE_ISDN
      "main" -> CommonDataKinds.Phone.TYPE_MAIN
      "otherFax" -> CommonDataKinds.Phone.TYPE_OTHER_FAX
      "radio" -> CommonDataKinds.Phone.TYPE_RADIO
      "telex" -> CommonDataKinds.Phone.TYPE_TELEX
      "ttyTdd" -> CommonDataKinds.Phone.TYPE_TTY_TDD
      "workMobile" -> CommonDataKinds.Phone.TYPE_WORK_MOBILE
      "workPager" -> CommonDataKinds.Phone.TYPE_WORK_PAGER
      "assistant" -> CommonDataKinds.Phone.TYPE_ASSISTANT
      "mms" -> CommonDataKinds.Phone.TYPE_MMS
      "other" -> CommonDataKinds.Phone.TYPE_OTHER
      else -> Columns.TYPE_CUSTOM
    }
    return phoneType
  }

  override fun getLabelFromCursor(cursor: Cursor): String {
    val label = super.getLabelFromCursor(cursor)
    return label
      ?: when (cursor.getInt(cursor.getColumnIndexOrThrow(Columns.TYPE))) {
        CommonDataKinds.Phone.TYPE_HOME -> "home"
        CommonDataKinds.Phone.TYPE_WORK -> "work"
        CommonDataKinds.Phone.TYPE_MOBILE -> "mobile"
        CommonDataKinds.Phone.TYPE_FAX_WORK -> "faxWork"
        CommonDataKinds.Phone.TYPE_FAX_HOME -> "faxHome"
        CommonDataKinds.Phone.TYPE_PAGER -> "pager"
        CommonDataKinds.Phone.TYPE_CALLBACK -> "callback"
        CommonDataKinds.Phone.TYPE_CAR -> "car"
        CommonDataKinds.Phone.TYPE_COMPANY_MAIN -> "companyMain"
        CommonDataKinds.Phone.TYPE_ISDN -> "isdn"
        CommonDataKinds.Phone.TYPE_MAIN -> "main"
        CommonDataKinds.Phone.TYPE_OTHER_FAX -> "otherFax"
        CommonDataKinds.Phone.TYPE_RADIO -> "radio"
        CommonDataKinds.Phone.TYPE_TELEX -> "telex"
        CommonDataKinds.Phone.TYPE_TTY_TDD -> "ttyTdd"
        CommonDataKinds.Phone.TYPE_WORK_MOBILE -> "workMobile"
        CommonDataKinds.Phone.TYPE_WORK_PAGER -> "workPager"
        CommonDataKinds.Phone.TYPE_ASSISTANT -> "assistant"
        CommonDataKinds.Phone.TYPE_MMS -> "mms"
        CommonDataKinds.Phone.TYPE_OTHER -> "other"
        else -> "unknown"
      }
  }
}
